### PR TITLE
Added - Last 10 articles can be sent through email.

### PR DESCRIPTION
**Pull request by MatiasComercio**
# Summary
- Last 10 articles can be sent through email. This action is triggered through a button on /articles view.

# Trello Card
https://trello.com/c/MzY1FopM/14-crear-email-con-los-ultimos-10-articulos-creados

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -102,3 +102,7 @@ gem 'devise'
 # creating elegant backends for website administration.
 gem 'activeadmin', github: 'activeadmin'
 gem 'inherited_resources', github: 'activeadmin/inherited_resources'
+
+# Dotenv: https://github.com/bkeepers/dotenv
+# Shim to load environment variables from .env into ENV in development.
+gem 'dotenv-rails', :groups => [:development, :test]
```
Comment (epintos) [inline Gemfile]: Its better to put this inside the developemtn and test group directly

File: app/views/layouts/application.html.slim [hunk 2]
```diff
@@ -11,9 +11,13 @@ html
...
-    / added due to devise suggestions
-    p.notice
-      = notice
-    p.alert
-      = alert
+    p.flash
+      - flash.each do |key, value|
+        span class=".alert.alert-#{key}"
+          = value
+    / added due to devise suggestions => replace with the above code
```
Comment (epintos) [inline app/views/layouts/application.html.slim]: Remove commented code if you don't use it

File: app/views/user_mailer/last_ten_articles.text.slim [hunk 3]
```diff
@@ -0,0 +1,16 @@
+' Hi
```
Comment (epintos) [inline app/views/user_mailer/last_ten_articles.text.slim]: Why `.text.slim`? Maybe its sth new in rails 5 or you named it like that?

File: config/environments/development.rb [hunk 4]
```diff
@@ -52,6 +47,25 @@
...
+  config.action_mailer.perform_caching = false
+
+  # **Own**
+  ## added as suggested in: https://github.com/plataformatec/devise
   config.action_mailer.default_url_options = { host: 'localhost', port: 3000 }
+
+  config.action_mailer.delivery_method = :smtp
+  config.action_mailer.smtp_settings = {
+    address: 'smtp.mailgun.org',
+    port: 587,
```
Comment (epintos) [inline config/environments/development.rb]: All this variables should go in secrets

File: config/routes.rb [hunk 5]
```diff
@@ -11,5 +11,9 @@
 
   get '/landing' => 'landing#index'
 
+  get 'send_last_ten_articles',
+      to: 'articles#send_last_ten_articles',
```
Comment (epintos) [inline config/routes.rb]: Better to use routes with the "resources" syntax

File: spec/fixtures/user_mailer/last_ten_articles [hunk 6]
```diff
@@ -0,0 +1,3 @@
+UserMailer#last_ten_articles
```
Comment (epintos) [inline spec/fixtures/user_mailer/last_ten_articles]: Missing the file extension

File: spec/mailers/user_mailer_spec.rb [hunk 7]
```diff
@@ -0,0 +1,25 @@
+require 'rails_helper'
+
+RSpec.describe UserMailer, type: :mailer do
+  describe 'last_ten_articles' do
+    let(:user) { create(:user) }
+    let(:from_mail) { 'no-reply@training.wolox.com.ar' }
+    let(:articles) { create_list(:article, 12) }
+    # let(:body_mail) do
+    #   body = 'Hi ' + user.email + '!'
```
Comment (epintos) [inline spec/mailers/user_mailer_spec.rb]: Commented code

File: spec/mailers/user_mailer_spec.rb [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+    let(:mail) { UserMailer.last_ten_articles user, Article.last(10) }
+
+    it 'renders the headers' do
+      expect(mail.subject).to eq('Keep up with the coolest Blog')
+      expect(mail.to).to eq([user.email])
+      expect(mail.from).to eq([from_mail])
+    end
+
+    # Don't really know if this is necessary or how to do it...
+    # it "renders the body" do
```
Comment (epintos) [inline spec/mailers/user_mailer_spec.rb]: Same

Comment (epintos) [inline spec/mailers/user_mailer_spec.rb]: I think i would try to test the variables with `@` like the views

File: app/views/user_mailer/last_ten_articles.text.slim (hunk 3 above)
Comment (MatiasComercioITBA) [inline app/views/user_mailer/last_ten_articles.text.slim]: This is the plain text version of the .html.slim file of the email. Should I remove it anyway?

File: config/routes.rb (hunk 5 above)
Comment (MatiasComercioITBA) [inline config/routes.rb]: What does it mean? I can't understand how should I change this.

File: spec/fixtures/user_mailer/last_ten_articles (hunk 6 above)
Comment (MatiasComercioITBA) [inline spec/fixtures/user_mailer/last_ten_articles]: Don't know which file extension is needed for this file as it has been autogenerated. I forgot to remove it; I have no idea if what is its purpose. Should I remove it?

File: app/views/user_mailer/last_ten_articles.text.slim (hunk 3 above)
Comment (epintos) [inline app/views/user_mailer/last_ten_articles.text.slim]: You can leave it, is okay

File: config/routes.rb (hunk 5 above)
Comment (epintos) [inline config/routes.rb]:
```
resources :articles do
   collection do
      get : send_last_ten_articles
   end
end
```

File: spec/fixtures/user_mailer/last_ten_articles (hunk 6 above)
Comment (epintos) [inline spec/fixtures/user_mailer/last_ten_articles]: Yeah.. if you don't use it remove it then

File: spec/mailers/user_mailer_spec.rb (hunk 8 above)
Comment (MatiasComercio) [inline spec/mailers/user_mailer_spec.rb]: I've found this: http://stackoverflow.com/questions/21080407/how-do-i-test-that-an-instance-variable-is-set-in-my-my-mailer-with-rspec#answer-21080501
What it basically says is that there is no way to test this with rspec, but that we can use the metaprogramming tools provided by ruby.
However, there is no way to achieve this here, as mail is a method call, not an instance of UserMailer.
That's why I, as we talked, have decided not to test here in this particular case.

**Comment by epintos** [comment]
👍

**Pull request merged**